### ISSUE TITLE
Finish moving CI to KinD

### ISSUE THREAD

**Issue by kdorosh** [solo-io/gloo#3299]
**Is your feature request related to a problem? Please describe.**
We need CI to scale (run multiple builds at once) without feeling pain. This may help with a glooctl check timeout flake we have seen a lot recently in CI.

**Describe the solution you'd like**
GitHub action to run CI regression tests against that run on different hosts so builds can be tested truly in parallel and against clean slate

**Additional context**
We need to make sure that releases still run the regression tests before a release, or ensure we always merge to master before a release.

**Comment by kdorosh** [comment]
Sample logs of flake referenced above https://github.com/solo-io/gloo/pull/3300#issuecomment-655713894

**Comment by EItanya** [comment]
> We need to make sure that releases still run the regression tests before a release, or ensure we always merge to master before a release.

We could also enable the setting which forces merging master into the PR branch before merging into master

**Comment by kdorosh** [comment]
closing as duplicate of https://github.com/solo-io/gloo/issues/2997
